t/job-manager/sched-helper.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+#
+
+# job-manager sched helper functions
+
+get_state() {
+	local id=$1
+	local state=$(${LIST_JOBS} | awk '$1 == "'${id}'" { print $2; }')
+	test -z "$state" \
+		&& flux job wait-event --timeout=5 ${id} clean >/dev/null \
+		&& state=I
+	echo $state
+}
+
+check_state() {
+	local id=$1
+	local wantstate=$2
+	for try in $(seq 1 10); do
+		test $(get_state $id) = $wantstate && return 0
+	done
+	return 1
+}
+
+get_annotation() {
+        local id=$1
+        local key=$2
+        local note="$(${LIST_JOBS} | grep ${id} | cut -f 6- | jq .\"${key}\")"
+        echo $note
+}
+
+check_annotation() {
+	local id=$1
+        local key=$2
+        local value="$3"
+	for try in $(seq 1 10); do
+		test "$(get_annotation $id $key)" = "${value}" && return 0
+                sleep 0.5
+	done
```

Comment:
Seems like we have some inconsistent tab/space usage going on (judging from the diff)